### DISCUSSION TITLE
[IntelliSense] Most class suggestions are prefixed with lf-

### DISCUSSION THREAD

**Discussion by erralb** [tailwindlabs/tailwindcss#9536] [Help] [answered]
**What version of Tailwind CSS IntelliSense are you using?**

v0.8.7

**What version of Tailwind CSS are you using?**

v3.1.8

**What package manager are you using?**

yarn within a Roots Sage 10 WP theme

**What operating system are you using?**

macOS, VSCode, latest version to date

**Reproduction URL**

Problem within VSCode

**Describe your issue**

Everything was working fine, but suddenly the class suggestions are prefixed with lf-

<img width="897" alt="Screenshot 2022-10-12 at 14 39 32" src="https://user-images.githubusercontent.com/790992/195344731-20d42aa1-87fd-4777-b6b3-0326b3a47470.png">

I did not change anything recently (new extension installation, configuration changes, etc.) and this started occurring. 
I checked the settings but could not find anything about it, if someone can help, cheers.

**Comment by bradlc** [comment]
Hey @erralb. It seems like `lf-` is the configured [prefix](https://tailwindcss.com/docs/configuration#prefix). Take a look in the output panel (`Tailwind CSS: Show Output`) and it will tell you which config file is being loaded. If not please share your project or a minimal one that reproduces the issue.

**Comment by erralb** [comment]
Hey, thanks for getting back.

Here is my `Tailwind CSS: Show Output` output :

```
Failed to load workspace modules.
Using bundled version of `tailwindcss`: v3.0.11
Using bundled version of `postcss`: v8.3.9

warn - The `purge`/`content` options have changed in Tailwind CSS v3.0.
warn - Update your configuration file to eliminate this warning.

warn - The `darkMode` option in your Tailwind CSS configuration is set to `false`, which now behaves the same as `media`.
warn - Change `darkMode` to `media` or remove it entirely.

warn - The RTL features in Tailwind CSS are currently in preview.
warn - Preview features are not covered by semver, and may be improved in breaking ways at any time.
Failed to load workspace modules.
Using bundled version of `tailwindcss`: v3.0.11
Using bundled version of `postcss`: v8.3.9
Failed to load workspace modules.
Using bundled version of `tailwindcss`: v3.0.11
Using bundled version of `postcss`: v8.3.9
```

Looks like my Tailwind config is not loaded by the VSCode extension.
I'm using [Sage 10](https://roots.io/sage/) with Tailwind. Tailwind is configured in Sage through a tailwind.config.cjs file. Here is the content if it can help : 

```
module.exports = {
  content: ["./index.php", "./app/**/*.php", "./resources/**/*.{php,vue,js}"],
  theme: {
    extend: {
      fontFamily: {
        'TTFirsNeue': ['TT Firs Neue'],
        'icomoon': ['icomoon'],
      },
      colors: {
        'button-purple': '#D296D5'
      },
      typography: (theme) => ({
        DEFAULT: {
          css: {
            h1: {
              'font-family': theme('fontFamily.TTFirsNeue'),
              'font-weight': '600',
              'text-transform': 'uppercase',
            },
            h2: {
              'font-family': theme('fontFamily.TTFirsNeue'),
              'font-size': '46px',
              'font-weight': '800',
              'text-transform': 'uppercase',
              'margin-top': '.5em',
              'margin-bottom': '.5em',
            },
            a: {
              '&:not(.go-to-next-section)': {
                'font-weight': 'bold',
                'text-decoration': 'underline',
                'color': '#AB8ED9',
                'font-style': 'italic'
              },
            },
            p: {
              'line-height': '1.5rem'
            },
          },
        },
      }),
    },
  },
  plugins: [
    require('@tailwindcss/typography'),
    require('@tailwindcss/forms'),
    function ({ addVariant }) {
      addVariant('child', '& > *');
      addVariant('child-hover', '& > *:hover');
    },
  ],
};

```
I've search the whole project for "lf-" but could not find anything.
Following your answer, I've tried adding an empty prefix to my Tailwind config, but since it looks like the extension is not loading it... As I said, it's weird, because it was working fine until today.

Is there a way for me to remove this prefix in the extension settings ?

**Comment by bradlc** [comment]
Very weird! Are you able to share your project or a minimal reproduction? Unfortunately it's not likely I'll be able to help debug this without that.

**Comment by erralb** [comment]
Hey, yes I could share it. How can I send it to you ?

One thing I just found out : If I load only the WP Theme folder into a new VSCode Workspace, the Tailwind config file is found.

`Tailwind CSS: Show Output `shows : 

```
Found Tailwind CSS config file: /Users/myusername/web/seer/web/app/themes/studio_seer/tailwind.config.cjs
Loaded postcss v8.4.14: /Users/myusername/web/seer/web/app/themes/studio_seer/node_modules/tailwindcss/node_modules/postcss
Loaded tailwindcss v3.1.8: /Users/myusername/web/seer/web/app/themes/studio_seer/node_modules/tailwindcss
```
And I do not have the problem anymore. 
But when the whole project folder is loaded into the VSCode workspace, I get a : 
```
Failed to load workspace modules.
Using bundled version of `tailwindcss`: v3.0.11
Using bundled version of `postcss`: v8.3.9
```

So I guess that's part of the problem.
Is there a way to specify in the settings where to find the Tailwind CSS config file per workspace ?

**Comment by bradlc** [comment, marked as answer]
If you could a GitHub repo that would be best 👍 

> _If I load only the WP Theme folder into a new VSCode Workspace, the Tailwind config file is found._

Based on this I'm fairly sure it will be because the extension is picking up a different config file than the one you expect. Is there a chance that there is a Tailwind config file in a folder that is excluded from VS Code search? Make sure you uncheck this toggle when searching for `lf-`

![CleanShot 2022-10-13 at 08 56 55@2x](https://user-images.githubusercontent.com/2615508/195536773-45be159e-52c1-4608-9246-cba58214cd87.png)

Having said that, you can force the extension to use a specific config file by using the `tailwindCSS.experimental.configFile` workspace setting:

```
"tailwindCSS.experimental.configFile": "path/to/tailwind.config.js"
```

**Reply by erralb** [reply]
Thanks, I didn't think about excluded files...
I found the culprit, the WordPress plugin LottieFiles uses Tailwind and adds the prefix lf- in its config file.

It would be useful if an alert was being displayed in case more than one Tailwind config file is detected, and then having the ability to choose the one to use for the current workspace.

Setting the `tailwindCSS.experimental.configFile` at workspace level fixed it, thank you very much for the support !

**Comment by bradlc** [comment]
I'm going to convert this to a discussion because I don't think it is an issue with the extension per se. However, we are working on improvements to config resolution that should allow you to avoid having to manually specify the config in cases like these.
